File: event-service/routes/eventRoutes.js
import { Router } from "express";
import eventController from "../controllers/eventController";

const eventRouter = Router();

eventRouter.get("/events", eventController.getAllEvents);
eventRouter.post("/events", eventController.createEvent);
eventRouter.get("/events/:id", eventController.getEventById);
eventRouter.patch("/events/:id", eventController.updateEvent);
eventRouter.delete("/events/:id", eventController.deleteEvent);

export default eventRouter;

File: event-service/routes/event_routes.go
package routes

import (
	"event-service/controllers"

	"github.com/gin-gonic/gin"
)

func SetUpRoutes(router *gin.RouterGroup) {
	router.GET("/events", controllers.GetAllEvents)
	router.POST("/events", controllers.CreateEvent)
	router.GET("/events/:id", controllers.GetEvent)
	router.PATCH("/events/:id", controllers.UpdateEvent)
	router.DELETE("/events/:id", controllers.DeleteEvent)
}
